### PR TITLE
feat(lab): use prettier, add pr lint ci, add lint rules

### LINKED ISSUE #106: Unconsistent code formatting
Currently there is auto code formatting on saving which is based on eslint rule but this behaviour is not work as expected when using prettier. See folowing pictures:
image

prettier result:
image

auto save result:
image

### PR DESCRIPTION
**Pull request by rihr**
Resolves #106

Prettier, prettierrc and .editorconfig is now used for formatting - the eslint extension will still show formatting errors in the editor.
Added more eslint rule extensions:
- @typescript-eslint/recommended
- @typescript-eslint/recommended-requiring-type-checking
- eslint-comments/recommended

Added linting CI in PRs

Fixed most lint errors and warnings according to new rules

### CODE REVIEW

**Comment by bertearazvan** [comment]
Does it make sense for this PR to trigger a release of the package?

**Comment by rihr** [comment]
> 
> 
> Does it make sense for this PR to trigger a release of the package?

Not really but the code does get changed

**Review by bertearazvan — APPROVED** [review]
Looks good 👍🏼

**Comment by bertearazvan** [comment]
> > Does it make sense for this PR to trigger a release of the package?
> 
> Not really but the code does get changed

Then maybe I would make it a `fix` instead

File: packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx [hunk 1]
```diff
@@ -23,9 +23,9 @@ import useStyles from './styles';
 // #endregion
 
 interface SnackbarState {
-  open?: boolean
-  message?: any
-  options?: SnackbarProps
+  open?: boolean;
+  message?: any;
```
[the remark this answers is not in the document]
Comment (rihr) [reply packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx:27]: I considered changing it but I don't know what the functionality of the snackbar is - up to @rosudhi

**Review by rosudhi — CHANGES_REQUESTED** [review]
All seems good.

**Review by rosudhi — APPROVED** [review]

**Review by rosudhi — CHANGES_REQUESTED** [review]
The idea is message will support not only string but html tag as well, hence please change into `String | ReactNode`  instead.

File: packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx [hunk 2]
```diff
@@ -51,21 +51,22 @@ const SnackbarProvider: React.FC<SnackbarProviderProps> = ({
...
+  const newAutoHideDuration =
+    state.options?.autoHideDuration || existingSnackbarProps.autoHideDuration;
 
-  const newTransitionComponent = state.options?.transitionComponent
-    || existingSnackbarProps.TransitionComponent;
+  const newTransitionComponent =
+    state.options?.transitionComponent ||
+    existingSnackbarProps.TransitionComponent;
 
   const showMessage = (message: string, options: SnackbarProps) => {
```
Comment (rosudhi) [inline packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx:61]: this also need to be changed as well

File: packages/react-components-lab/src/components/Snackbar/types.ts [hunk 3]
```diff
@@ -4,13 +4,13 @@ import { ReactNode } from 'react';
...
-interface SnackbarContexValue {
+interface SnackbarContextValue {
   /**
    * Display a message with this snackbar.
-   * @param {any} message message to display
+   * @param {string} message message to display
    * @param {object} options options parameters that will be passed to the snackbar renderer
    */
-  showMessage(message: any, options?: SnackbarProps): void
+  showMessage(message: string, options?: SnackbarProps): void;
```
Comment (rosudhi) [inline packages/react-components-lab/src/components/Snackbar/types.ts:13]: this also need to be changed as well

**Review by rosudhi — CHANGES_REQUESTED** [review]
Forgot to add comments in specific line that need to be changed

File: .github/workflows/pr-lab.yml [hunk 4]
```diff
@@ -1,9 +1,8 @@
-name: CI - lab
+name: CI - PR Lint
```
Comment (bertearazvan) [inline .github/workflows/pr-lab.yml:1]:
```suggestion
name: CI - Lab PR Lint
```

File: .github/workflows/pr-lab.yml [hunk 5]
```diff
@@ -1,9 +1,8 @@
...
 on:
-  push:
-    pull_request:
-      types: [opened, synchronize, reopened, closed]
-      branches:
+  pull_request:
+    types: [opened, synchronize, edited]
+    branches:
```
Comment (bertearazvan) [inline .github/workflows/pr-lab.yml:5]: I am not sure if this is correct, but I will count on your testing. Not sure if `push` should be removed.

**Review by bertearazvan — COMMENTED** [review]
I can still see many `any` for `msg` or so. Shouldn't those be also `string | ReactNode`?

**Comment by rosudhi** [comment]
so far i can see, there are two line more, they are in SnackProvider.tsx file.
1.  Change into `String | ReactNode`
image
2.  i think for this one, we can just remove it.
image

**Review by rosudhi — APPROVED** [review]
Perfect!

**Pull request merged**